From c5992fcf1ffbbca525b4752b000cbef35fe3c761 Mon Sep 17 00:00:00 2001
From: Pavel Hrdina <phrdina@redhat.com>
Date: Mon, 3 Oct 2016 16:51:20 +0200
Subject: [PATCH] virtinst.Interface: fix interface creation with child
 interfaces

Instead of hard-coding every child interface as type='ethernet' use an
existing XML definition from libvirt if possible.  This fixes a bug
where bridge can be configured with bond interface which contain child
interfaces too.

RHEL-only

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=1350683

Signed-off-by: Pavel Hrdina <phrdina@redhat.com>
---
 virtinst/Interface.py | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/virtinst/Interface.py b/virtinst/Interface.py
index f1ae0be..1df8659 100644
--- a/virtinst/Interface.py
+++ b/virtinst/Interface.py
@@ -301,10 +301,9 @@ class _InterfaceCompound(Interface):
         xml = ""
         for i in self.interfaces:
             if type(i) is str:
-                name = i
+                iface_xml = "    <interface name='%s' type='ethernet'/>\n" % i
             else:
-                name = i.name()
-            iface_xml = "    <interface name='%s' type='ethernet'/>\n" % name
+                iface_xml = i.XMLDesc(libvirt.VIR_INTERFACE_XML_INACTIVE)
 
             xml += iface_xml
         return xml
-- 
2.10.0

